store/sqlstore/post_store.go

```diff
@@ -1373,3 +1373,42 @@ func (s *SqlPostStore) GetRepliesForExport(parentId string) store.StoreChannel {
 		}
 	})
 }
+
+func (s *SqlPostStore) GetDirectPostParentsForExportAfter(limit int, afterId string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		var posts []*model.DirectPostForExport
+		_, err1 := s.GetSearchReplica().Select(&posts, `
+                SELECT
+                    p1.*,
+                    Users.Username as Username,
```

Comment:
I'd suggest doing the same split query strategy to fetch the members as above to start, eliminating the `GROUP_CONCAT`. Odds are high that you'll be dealing with duplicate channel ids (given the likelihood of posts from the same channel appearing), so if you deduplicate before doing the `ChannelId IN (...)`, it might be a net win in terms of performance.